src/GAP/iso_gap_oscar.jl

```diff
@@ -55,13 +59,26 @@ function _iso_gap_oscar_field_rationals(F::GAP.GapObj)
    return MapFromFunc(x -> fmpq(x), x -> GAP.Obj(x), F, QQ)
 end
 
+function _iso_gap_oscar_ring_integers(F::GAP.GapObj)
+   return MapFromFunc(x -> fmpz(x), x -> GAP.Obj(x), F, ZZ)
```

Comment:
Actually, that's an unnecessary level of indirection (same for the QQ code a few lines above -- all my fault, of course)
```suggestion
   return MapFromFunc(fmpz, GAP.Obj, F, ZZ)
```

Reply:
No.
`MapFromFunc` expects two `Function` arguments, but `fmpz` and `GAP.Obj` have type `DataType`.

Reply:
Ah, that sounds like a deficiency in `MapFromFunc` then.

Reply:
@fingolfin see thofma/Hecke.jl/pull/592